TensorLib/Basic.lean

```diff
@@ -2,4 +2,98 @@ namespace TensorLib
 
 def hello := "world"
 
+class TensorElement (a: Type) where
+  add: a -> a -> a
+  -- etc. enough to implement numpy ops
+
+inductive TensorData (a: Type) [TensorElement a] where
+| td_root (x: List a)
+| td_node (tl: List (TensorData a))
+
+-- Awkward... I'm sure this exists somewhere in the stdlib
+def sequence {a: Type} (x: List (Option a)): Option (List a) :=
```

Comment:
You need to familiarize yourself with monads in Lean, this kind of code will be less Awkward